docs/technical_specs/core/util/tags.rst

```diff
@@ -0,0 +1,109 @@
+Tagging Classes
+===============
+
+IDAES contains classes for tagging model quantities and grouping them.  The tags
+allow a convenient short cut to important model inputs and outputs.  There are
+also facilities for numeric formatting and displaying output in desired units.
+
+Examples:
+---------
+
+The code below creates a simple model that can be used to demonstrate the use
+of tags.
+
+.. testcode::
+
+  import pyomo.environ as pyo
+  from idaes.core.util import ModelTag, ModelTagGroup
+
+  def model():
+      m = pyo.ConcreteModel()
+      m.w = pyo.Var([1, 2, 3], ["a", "b"], initialize=4, units=pyo.units.kg)
+      m.x = pyo.Var([1, 2, 3], initialize=5, units=pyo.units.kg)
+      m.y = pyo.Var(initialize=6, units=pyo.units.s)
+      m.z = pyo.Var(initialize=7)
+      m.e = pyo.Expression(expr=m.w[1, "a"] / m.x[1])
+      m.f = pyo.Expression(expr=m.x[1] / m.y)
+
+      @m.Expression([1, 2, 3], ["a", "b"])
+      def g(b, i, j):
+          return m.w[i, j] / m.x[i] * 100
+
+      return m
+
+The next code snippet creates a single tag object for the model variable ``w``.
+While the structure of the example model is simple, IDAES models often have a
+complex structure. Tags generally provide a much shorter means to reference a
```

Comment:
Another minor comment: I would replace the final "." with ", and " to combine the final statement in to the previous sentence for better flow.